File: langhuan-service/src/main/java/com/langhuan/common/GlobalExceptionHandler.java
package com.langhuan.common;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.security.access.AccessDeniedException;
import org.springframework.security.authorization.AuthorizationDeniedException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.context.request.WebRequest;

@Slf4j
@RestControllerAdvice
public class GlobalExceptionHandler {


    @ExceptionHandler(Exception.class)
    public Result<?> handleException(Exception e, HttpServletRequest request) {
        log.error("System Exception: uri={}", request.getRequestURI(), e);
        return Result.error(500, "System Error");
    }

    @ExceptionHandler(BusinessException.class)
    public Result<?> handleBusinessException(BusinessException e, HttpServletRequest request) {
        log.error("Business Exception: uri={}, code={}, message={}", request.getRequestURI(), e.getCode(), e.getMessage());
        return Result.error(e.getCode(), e.getMessage());
    }

    @ExceptionHandler(BaseException.class)
    public Result<?> handlerGlobalException(HttpServletResponse response, BaseException e) {
        log.error("BaseException请求异常：", e);
        response.setStatus(e.getResponseCode().getCode());

        return Result.error(e.getResponseCode().getCode(), e.getResponseCode().getMessage());
    }

    @ExceptionHandler(AccessDeniedException.class)
    public Result<?> handlerGlobalException(Exception e) {
        log.error("AccessDeniedException请求异常：", e);
        return Result.error(401, e.getMessage());

    }

    @ExceptionHandler(AuthorizationDeniedException.class)
    public Result<?> handleAuthorizationDeniedException(AuthorizationDeniedException ex) {
        log.error("AuthorizationDeniedException请求异常：", ex);
        return Result.error(HttpStatus.FORBIDDEN.value(), ex.getMessage());
    }

}

File: langhuan-server/src/main/kotlin/com/langhuan/common/GlobalExceptionHandler.kt
package com.langhuan.common

import io.jsonwebtoken.ExpiredJwtException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatus
import org.springframework.security.access.AccessDeniedException
import org.springframework.security.authorization.AuthorizationDeniedException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.context.request.WebRequest

@RestControllerAdvice
class GlobalExceptionHandler {
    
    companion object {
        private val log = LoggerFactory.getLogger(GlobalExceptionHandler::class.java)
    }

    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception, request: HttpServletRequest): Result<Any> {
        log.error("System Exception: uri={}", request.requestURI, e)
        return Result.error(500, "System Error")
    }

    @ExceptionHandler(BusinessException::class)
    fun handleBusinessException(e: BusinessException, request: HttpServletRequest): Result<Any> {
        log.error("Business Exception: uri={}, code={}, message={}", request.requestURI, e.code, e.message)
        return Result.error(e.code, e.message)
    }

    @ExceptionHandler(BaseException::class)
    fun handlerGlobalException(response: HttpServletResponse, e: BaseException): Result<Any> {
        log.error("BaseException请求异常：", e)
        response.status = e.responseCode?.code ?: 500

        return Result.error(e.responseCode?.code ?: 500, e.responseCode?.message ?: "Unknown Error")
    }

    @ExceptionHandler(AccessDeniedException::class)
    fun handlerGlobalException(e: Exception): Result<Any> {
        log.error("AccessDeniedException请求异常：", e)
        return Result.error(401, e.message ?: "Access Denied")
    }

    @ExceptionHandler(AuthorizationDeniedException::class)
    fun handleAuthorizationDeniedException(ex: AuthorizationDeniedException): Result<Any> {
        log.error("AuthorizationDeniedException请求异常：", ex)
        return Result.error(HttpStatus.FORBIDDEN.value(), ex.message ?: "Authorization Denied")
    }

    @ExceptionHandler(TokenExpiredException::class)
    fun handleTokenExpiredException(ex: TokenExpiredException): Result<Any> {
        log.error("TokenExpiredException请求异常：", ex)
        return Result.error(ex.responseCode!!.code, ex.responseCode!!.message)
    }

    @ExceptionHandler(ExpiredJwtException::class)
    fun handleExpiredJwtException(ex: ExpiredJwtException): Result<Any> {
        log.error("ExpiredJwtException请求异常：", ex)
        return Result.error(ResponseCodeEnum.UNAUTHORIZED.code, "Token已过期，请重新登录")
    }
}
